xbmc/guilib/GUIControlFactory.cpp

```diff
@@ -490,22 +490,23 @@ bool CGUIControlFactory::GetAnimations(TiXmlNode *control, const CRect &rect, in
   return ret;
 }
 
-bool CGUIControlFactory::GetActions(const TiXmlNode* pRootNode, const char* strTag, CGUIAction& action)
+bool CGUIControlFactory::GetActions(const TiXmlNode* pRootNode,
+                                    const char* strTag,
+                                    CGUIAction& actions)
 {
-  action.m_actions.clear();
+  actions.Reset();
   const TiXmlElement* pElement = pRootNode->FirstChildElement(strTag);
   while (pElement)
   {
     if (pElement->FirstChild())
     {
-      CGUIAction::cond_action_pair pair;
-      pair.condition = XMLUtils::GetAttribute(pElement, "condition");
-      pair.action = pElement->FirstChild()->Value();
-      action.m_actions.push_back(pair);
+      CGUIAction::CExecutableAction action = CGUIAction::CExecutableAction(
+          XMLUtils::GetAttribute(pElement, "condition"), pElement->FirstChild()->Value());
```

Comment:
Shorter: `const CGUIAction::CExecutableAction action{XMLUtils::GetAttribute(pElement, "condition"), pElement->FirstChild()->Value()};`

Reply:
This is just a style (apart from the missing const specifier) thing right? 
I don't see where the provided alternative is any different from what I'm already doing (hence I don't understand the `Prefer ctor with params over default ctor, followed by assigment` unless you're just providing an overall comment)

Reply:
what you wrote means:

1) default construct `CGUIAction::CExecutableAction`
2) construct a CGUIAction::CExecutableAction with the given params
3) call (the compiler-generated) CGUIAction::CExecutableActionoperator= to copy the data from the second instance to the first

Yes, modern compilers can optimize this, but I tend never to rely on such magic.

What I propose menas:

1) construct a CGUIAction::CExecutableAction with the given params

done.

Reply:
_confused_ 
i was of the strong opinion what enen92 had there is copy initialization (neither default ctor nor operator= involved, because it is not an assignment but an init) and your proposal is direct-list-initialization that falls back to the appropiate ctor (since we have no ctor `CGUIAction::CExecutableAction(std::initializer_list<>)`)...

Reply:
> what enen92 had there is copy initialization

 my point is to use direct initialization, it's cheaper (and the code is very clear about what happens).